pymeasure/experiment/results.py

```diff
@@ -138,12 +149,54 @@ def format(self, record):
         :type record: dict
         :return: a string
         """
-        return self.delimiter.join('{}'.format(record[x]) for x in self.columns)
+        return self.delimiter.join(
+            '{}'.format(record[x]) for x in self.columns)
 
     def format_header(self):
         return self.delimiter.join(self.columns)
 
 
+class CSVFormatter_Pandas(Results_Formatter):
+    """ Formatter of data results, pandas dataframe or single-line CSV """
+
+    def format(self, record):
+        """Formats a record as csv using pandas built-in ``to_csv`` method.
+        Accepts a pandas dataframe or a dict of values matching
+        the given list of columns.
+
+        :param record: record to format.
+        :type record: pandas.DataFrame or dict
+        :return: str
+        """
+        if isinstance(record, pd.DataFrame):
+            pass
+        elif isinstance(record, dict):
+            # ensure correct order of columns
+            record = (record[c] for c in self.columns)
+            record = pd.DataFrame(record)
+        else:
+            log.error('Formatting of data failed. ' +
+                      'Pandas dataframe or dict required.')
+            return ''
```

Comment:
This may be a too gentle way to handle a wrong type of data provided; this way, the measurement will continue without storing any data if it is provided as a wrong type.

I suggest changing it to the following:
```suggestion
            raise TypeError('Formatting of data failed. '
                            'Pandas dataframe or dict required.')
```